compiler/src/dotty/tools/dotc/transform/PatternMatcher.scala

```diff
@@ -791,12 +791,6 @@ object PatternMatcher {
      *  plan and following onSuccess plans.
      */
     private def collectSwitchCases(plan: TestPlan): List[Plan] = {
-      def isSwitchableType(tpe: Type): Boolean =
-        (tpe isRef defn.IntClass) ||
-        (tpe isRef defn.ByteClass) ||
-        (tpe isRef defn.ShortClass) ||
-        (tpe isRef defn.CharClass)
-
```

Comment:
I tried with `isSwitchableType` and it seems to work if you widen before calling it: 
```scala
isSwitchableType(scrutinee.tpe.widen)
```

Reply:
done